iDiff/utils/tar_utils.go

```diff
@@ -15,6 +15,32 @@ import (
 	"github.com/docker/docker/pkg/system"
 )
 
+// ImageToDir converts an image to an unpacked tar and creates a representation of that directory.
+func ImageToDir(img string) (string, string, error) {
+	cli, err := client.NewEnvClient()
+	if err != nil {
+		return "", "", err
+	}
+	tarPath, err := ImageToTar(cli, img)
+	if err != nil {
+		return "", "", err
+	}
+	//tarPath := filepath.Join(img, ".tar")
+	err = ExtractTar(tarPath)
+	if err != nil {
+		return "", "", err
+	}
+	os.Remove(tarPath)
+	path := strings.TrimSuffix(tarPath, filepath.Ext(tarPath))
+	jsonPath := path + ".json"
```

Comment:
We had a problem that using `filepath.Join()` when concatenating file extension types it created `thing/.tar` or `thing/.json` instead of just `thing.tar` and `thing.json`.  This resulted in more problems of trying to load the files in other functions.  Do you know of a way around that problem?